ibis/expr/rules.py

```diff
@@ -359,12 +359,17 @@ def output_dtype(self):
 
 
 @public
-def shape_like(name):
+def shape_like(*names):
     @attribute.default
     def output_shape(self):
-        args = getattr(self, name)
-        args = args if util.is_iterable(args) else [args]
-        return highest_precedence_shape(args)
+        input_args = []
+        for name in names:
+            args = getattr(self, name)
+            if util.is_iterable(args):
```

Comment:
Could you please implement it directly in the op since it is required for a single operation only.

Reply:
Done.